src/thunderbots/software/ai/navigator/obstacle.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include "geom/util.h"
+#include "ai/world/world.h"
+
+constexpr double DEFAULT_AVOID_DIST = 0.15;  // meters
+
+class RobotObstacle
+{
+   public:
+    /**
+     * Constructor.
+     *
+     * @param r Robot to create obstacle from.
+     */
+    RobotObstacle(Robot& r, double avoid_dist);
+
+    /**
+     * Returns the violation distance of the given point inside the obstacle
+     * boundary; 0.0 if no violation.
+     *
+     * @param p point to check if violating
+     *
+     * @return Violation distance, defined as distance from the nearest boundary
+     *         to the given point; 0.0 if no violation occured.
+     */
+    double getViolation(Point& p);
```

Comment:
Hmm, I would think `std::optional` is a bit overkill here, since a violation distance of 0.0 makes sense in this case. I don't see it as a "special" value, as it means that the point is 0.0 metres inside the boundary (i.e. at or outside the obstacle boundary).

Reply:
OK, I see what you mean now. On another note I think naming this `getViolationDistance` would be a bit more clear.

Reply:
Renamed to `getViolationDistance`